recipes/default.rb

```diff
@@ -14,10 +14,18 @@
 # limitations under the License.
 #
 
+ruby_block 'ensure node can resolve API FQDN' do
+  extend ChefAnalyticsCookbook::Helpers
+  block { repair_api_fqdn }
+  only_if { api_fqdn_available? }
+  not_if { api_fqdn_resolves? }
+end
```

Comment:
this is going to create a race-to-lazy-everything.

any future resource that uses `node['chef-analytics']['api_fqdn']` must now reference it as `lazy { node['chef-analytics']['api_fqdn'] }` and no resource that comes before this in the run_list can possibly get the correct value.

if the attempt is to lazy this so that it can repair the value from a following cookbook which sets it in compile mode, then the appropriate fix for that issue is that the subsequent cookbook should set this value in its attribute file (that cookbook should be more eager instead of making this cookbook more lazy).